src/components/ExchangePage/index.jsx

```diff
@@ -235,14 +242,42 @@ function getMarketRate(
   }
 }
 
-export default function ExchangePage({ initialCurrency, sending }) {
+export default function ExchangePage({ initialCurrency, sending = false, params }) {
   const { t } = useTranslation()
   const { account } = useWeb3Context()
 
+  const [, toggleDarkMode] = useDarkModeManager()
+
+  useEffect(() => {
+    if (params.theme) {
+      toggleDarkMode(
+        params.theme === SUPPORTED_THEMES.DARK ? true : params.theme === SUPPORTED_THEMES.LIGHT ? false : ''
+      )
+    }
+  }, []) // eslint-disable-line react-hooks/exhaustive-deps
+
   const addTransaction = useTransactionAdder()
 
-  const [rawSlippage, setRawSlippage] = useState(ALLOWED_SLIPPAGE_DEFAULT)
-  const [rawTokenSlippage, setRawTokenSlippage] = useState(TOKEN_ALLOWED_SLIPPAGE_DEFAULT)
+  // check if URL specifies valid slippage, if so use as default
+  const initialSlippage = (token = false) => {
+    let slippage = parseFloat(params.slippage)
+    if (Number(slippage) === 0 || Number(slippage) >= 1) {
+      return Math.round(slippage) // round to match custom input availability
+    }
+    // check for token <-> token slippage option
+    return token ? TOKEN_ALLOWED_SLIPPAGE_DEFAULT : ALLOWED_SLIPPAGE_DEFAULT
+  }
+
+  // check URL params for recipient, only on send page
+  const initialRecipient = () => {
+    if (sending && params.recipient) {
+      return params.recipient
+    }
+    return ''
+  }
+
+  const [rawSlippage, setRawSlippage] = useState(initialSlippage())
+  const [rawTokenSlippage, setRawTokenSlippage] = useState(initialSlippage(true))
```

Comment:
this approach looks ok, but there are some rough edges. `params.slippage` will be a string of an integer, so we should use `Number.parseInt`, not `parseFloat`. Then, you'll need to check that the parsed value isn't `NaN`. The rounding looks a little wonky as well, the function should just return an int from `0` to whatever we decided our max was (`200`?).

Bringing this up because for example [https://deploy-preview-423--uniswap.netlify.com/swap?slippage=10](https://deploy-preview-423--uniswap.netlify.com/swap?slippage=10) doesn't work.

Reply:
also, you can pass a lazy init function to `useState` so that it runs only on mount: `() => initialSlippage()`. minor perf gains but still probably worth it everywhere you use this pattern!

Reply:
Yep added in the integrer parsing instead. I also got rid of the rounding because we're good once we check thats its an integer - good idea

The bounds for this entry we want are now is bips so the acceptable range is 0->9999 (because they can enter anything with typing from 0%->99.99%) - warning populate if they put in a value thats outside our acceptable range 

That deploy was actually broken for a different reason which i just fixed too. Basically our button list wasn't accounting for the case that a user entered a custom slippage amount that was the same as one of the buttons. Right now if they enter 100 it will select the 1% button (same with 10 and 50) <- we can change this so that entering 100 will fill in the custom field instead

- good idea with the lazy init too